src/reflect/scala/reflect/internal/Printers.scala

```diff
@@ -562,7 +562,7 @@ trait Printers extends api.Printers { self: SymbolTable =>
 
       def addBackquotes(s: String) =
         if (decoded && (decName.exists(ch => brackets.contains(ch) || isWhitespace(ch) || isDot(ch)) ||
-          (name.isOperatorName && decName.exists(isOperatorPart) && decName.exists(isScalaLetter) && !decName.contains(bslash))))
+          (name.isOperatorName && decName.exists(isOperatorPart) && decName.exists(isScalaLetter) && !decName.contains(bslash)) || (decName.exists(isOperatorPart) && decName.exists(isScalaLetter) && !decName.contains(bslash))))
```

Comment:
It looks like you just duplicated the expression and dropped one condition, which is redundant.

Is the `isOperatorName` test too strict? Also I'm not sure why the issue says it's OK to supply an unencoded name.

Reply:
`name.isOperatorName` is true if name contains operator names like `$plus`, `$times`, `$minus`  when decoded the operator name becomes a operator (e.g `$plus` becomes `+`). 
The condition `(name.isOperatorName && decName.exists(isOperatorPart) && decName.exists(isScalaLetter) && !decName.contains(bslash))` will only be true for names like `x$plusy`.

 If name is already decoded like `x-y` then this condition will fail and the generated code is not valid. To make it produce valid code I had to omit the name.isOperatorName.

```scala
scala> print(showCode(q"""val ${TermName("x$plusy")} = 1"""))
val `x+y` = 1
scala> print(showCode(q"""val ${TermName("x-y")} = 1"""))
val `x-y` = 1
```

I could have written
` ((name.isOperatorName || decName.exists(isOperatorPart)) && decName.exists(isScalaLetter) && !decName.contains(bslash))` instead of duplicating it which is better. Let me change it.